### DISCUSSION TITLE
Open a table HTML with visidata

### DISCUSSION THREAD

**Discussion by nvallinoto** [saulpw/visidata#2060] [Q&A] [answered]
Hi all,
this is my first message in this community so I would like to send a very big thanks for this extraordinary application.
I have a trouble searching to grab data from the table in the following url
https://unfccc.int/process/parties-non-party-stakeholders/non-party-stakeholders/admitted-ngos/list-of-admitted-ngos
Depends on html ?
Thanks a lot,
Nicola

**Comment by aborruso** [comment]
Ciao Nicola,
if you try simply curl that website it seems almost empty.

If you download it, waiting for the dom rendering you will able to open that table, because you will have it in the code. I use a lot the LINUX cli:

```
google-chrome-stable --headless --disable-gpu --print-to-stdout --virtual-time-budget=10000 --dump-dom https://unfccc.int/process/parties-non-party-stakeholders/non-party-stakeholders/admitted-ngos/list-of-admitted-ngos  >nic.html
```

And then I run vd

![image](https://github.com/saulpw/visidata/assets/30607/30844cf8-61f8-49b4-84d4-d493edeb816d)

![image](https://github.com/saulpw/visidata/assets/30607/c60790ec-b84b-4a76-8810-60fb3ebe45c7)

I do not know if there is a VD option to wait for the DOM rendering

**Reply by nvallinoto** [reply]
thanks Andrea for your quick answer and for sending me the file. I'm using visidata in windows.

**Reply by aborruso** [reply, marked as answer]
> thanks Andrea for your quick answer and for sending me the file. I'm using visidata in windows.

Me too. But I work in Linux CLI using WSL for Windows.

If you want to use windows prompt, you can run the same command, using chrome for win.
Something like


```
"C:\Program Files\Google\Chrome\Application\chrome.exe" --headless --disable-gpu --virtual-time-budget=10000 --dump-dom --no-sandbox "https://unfccc.int/process/parties-non-party-stakeholders/non-party-stakeholders/admitted-ngos/list-of-admitted-ngos" > tmp.html
```

This is the output I have had and it contains you table: https://gist.github.com/aborruso/ac824473cb66b0bbadecc91e741c7dc0

**Reply by nvallinoto** [reply]
I worked also for me. Thanks a lot. 
Now I can work to clean the data.
Have a good day Andrea.
Nicola

**Reply by aborruso** [reply]
I'm happy @nvallinoto 

So you can mark this answer as a solution to your problem :)

**Reply by aborruso** [reply]
Dear @saulpw is there a vd option to wait for the full rendering of the dom, when trying to open a web page?

Thank you

**Reply by saulpw** [reply]
@aborruso No...I like your solution though!  I'll be using it myself sometimes.

**Comment by aborruso** [comment]
@nvallinoto another way is simply to save the page via browser.

The output will contain the rendered html table

![image](https://github.com/saulpw/visidata/assets/30607/6c132a9e-d5ad-44ec-83e4-ba172bb4bef0)

**Reply by nvallinoto** [reply]
oh yes! Simpler.

**Reply by aborruso** [reply]
But in the other way you can make a programmatic use: a script that downloads and normalizes the data every 3 months at 12:00 noon

**Reply by nvallinoto** [reply]
yes this is true.
